### DISCUSSION TITLE
HW 3

### DISCUSSION THREAD

**Discussion by ElisDiaz46** [cooperrc/computational-mechanics#167] [Q&A]
npz_coords = np.load('../data/projectile_coords.npz')
t = npz_coords['t']
x = npz_coords['x']
y = npz_coords['y']

I have done this and also done this 

np.polyfit(x, y, t, rcond=None, full=False, w=None, cov=False)

But, I am getting this Error

**Comment by cooperrc** [comment]
> `npz_coords = np.load('../data/projectile_coords.npz')`
Do you get an error after this line?

> t = npz_coords['t'] 
> x = npz_coords['x'] 
> y = npz_coords['y']

These will work only after that first line worked. 


> I have done this and also done this
> 
> np.polyfit(x, y, t, rcond=None, full=False, w=None, cov=False)

`np.polyfit` takes 3 arguments: `np.polyfit(t, x, order)`
- `t` is the independent variable, in this case, time
- `x` is the dependent variable, in this case either x-position or y-position (not both at the same time)
- `order` is the polynomial power, e.g. `2` for a quadratic equation and `1` for a linear equation

**Reply by ElisDiaz46** [reply]
No, I don't get an error after this line: npz_coords = np.load('../data/projectile_coords.npz')

I got it after np.polyfit(x, y, t, rcond=None, full=False, w=None, cov=False)

I fix the issue 

However i have another issue. I don't know for the first question if this out put is the acceleration for this first order line:

Input: 
np.polyfit(t,x, 1, rcond=None, full=False, w=None, cov=False)

Output: 
array([37.67821278,  1.79405824])

**Reply by cooperrc** [reply]
> Input: np.polyfit(t,x, 1, rcond=None, full=False, w=None, cov=False)
> 
> Output: array([37.67821278, 1.79405824])

This gives you $x(t) = 37.67 t + 1.79$ as the best-fit for your data. So, the _assumed_ acceleration was 0 and the speed is 37.67 m?/s

**Reply by ElisDiaz46** [reply]
How would we get the acceleration from this equation , p.polyfit(t,x, 1, rcond=None, full=False, w=None, cov=False)? 
would have to put this p.polyfit(t,x, 1, rcond=None, full=False, w=None, cov=False) in 2 nd order?
